deeplabcut/pose_estimation_tensorflow/predict_supermodel.py

```diff
@@ -0,0 +1,673 @@
+import argparse
+import os
+import os.path
+import pickle
+import re
+import time
+from pathlib import Path
+
+import cv2
+import imgaug.augmenters as iaa
+import numpy as np
+import pandas as pd
+from skimage.util import img_as_ubyte
+from sklearn.metrics.pairwise import cosine_similarity
+from tqdm import tqdm
+
+from deeplabcut.pose_estimation_tensorflow.config import load_config
+from deeplabcut.pose_estimation_tensorflow.core import \
+    predict as single_predict
+from deeplabcut.pose_estimation_tensorflow.core import \
+    predict_multianimal as predict
+from deeplabcut.pose_estimation_tensorflow.lib import inferenceutils
+from deeplabcut.pose_estimation_tensorflow.lib import trackingutils
+from deeplabcut.utils import auxfun_multianimal
+from deeplabcut.utils import auxiliaryfunctions
+from deeplabcut.utils.auxfun_videos import VideoWriter
+
+
+# instead of having these in a lengthy function, I made this a separate function
+def get_nuances(
```

Comment:
Can we also reuse this function (which should probably be renamed, it's not very descriptive) elsewhere?

Reply:
You mean in other part of DLC code? Yes we should.